framework/log/Logger.php

```diff
@@ -115,13 +115,11 @@ class Logger extends Component
     public function init()
     {
         parent::init();
-        register_shutdown_function(function () {
-            // make regular flush before other shutdown functions, which allows session data collection and so on
-            $this->flush();
-            // make sure log entries written by shutdown functions are also flushed
-            // ensure "flush()" is called last when there are multiple shutdown functions
-            register_shutdown_function([$this, 'flush'], true);
-        });
+        // make regular flush before other shutdown functions, which allows session data collection and so on
```

Comment:
maybe the session shutdown should be aware of this and register on a later priority.

In general we may introduce constants e.g. `Session::SHUTDOWN_PRIORITY = 50` , which can be used to register shutdown functions before some others, e.g. 

```php
// do something before session is closed
Yii::registerShutdownFunction($callback, [], Session::SHUTDOWN_PRIORITY - 1);
```